### PR TITLE
chore(DATAGO-124293): add CLI task commands (send and run)

### PR DESCRIPTION
**Pull request by efunneko**
### What is the purpose of this change?
Adds two new CLI commands for closed-loop testing of SAM without a browser:

1. **`sam task send`** - Send a task to a running SAM instance and stream the response
2. **`sam task run`** - Start SAM, send a task, stream response, and stop - all in one command

These commands enable rapid iteration and debugging by allowing developers to test agent behavior directly from the terminal.

Related: https://sol-jira.atlassian.net/browse/DATAGO-124293

### How was this change implemented?

**New files for `sam task send`:**
- `cli/commands/task_cmd/send.py` - Main send command
- `cli/commands/task_cmd/sse_client.py` - SSE streaming client
- `cli/commands/task_cmd/message_assembler.py` - Event processing and text assembly
- `cli/commands/task_cmd/event_recorder.py` - YAML event recording
- `cli/commands/task_cmd/artifact_handler.py` - Artifact downloads

**New files for `sam task run`:**
- `cli/commands/task_cmd/run.py` - Main run command combining SAM lifecycle + task
- `cli/commands/task_cmd/sam_runner.py` - SAM lifecycle manager using `SolaceAiConnector`
- `cli/commands/task_cmd/common.py` - Shared utilities for both commands

**Modified files:**
- `cli/commands/task_cmd/__init__.py` - Register both commands
- `docs/docs/documentation/components/cli.md` - Documentation for both commands

**Key features:**
- SSE streaming for real-time response output
- Artifact and STIM file downloads to output directory
- Session continuity support (`--session-id`)
- Agent readiness polling with stabilization delay
- Guaranteed cleanup via context manager pattern

### Key Design Decisions

1. **Two-command approach**: `send` for use with separately running SAM, `run` for self-contained testing

2. **Readiness detection with stabilization**: After first detecting agents, waits 2 seconds and polls again to avoid race conditions

3. **SAMRunner context manager**: Ensures cleanup is always called, even on errors or interrupts

4. **Output directory structure**: 
   ```
   /tmp/sam-task-{taskId}/
   ├── response.txt      # User-facing text
   ├── sse_events.yaml   # All SSE events
   ├── {taskId}.stim     # STIM file
   └── artifacts/        # Downloaded artifacts
   ```

### How was this change tested?
- [x] Manual testing: `sam task send "Hello"` with running SAM
- [x] Manual testing: `sam task run "What agents are available?"` 
- [x] Verified SSE streaming works correctly
- [x] Verified artifact downloads
- [x] Verified clean shutdown on completion
- [x] Known limitations: No automated unit tests added yet

### Is there anything the reviewers should focus on?
- The SSE client implementation for edge cases
- Agent readiness polling algorithm in `wait_for_agents()`
- Error handling during SAM startup/shutdown

### CODE REVIEW

File: cli/commands/task_cmd/run.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        _info(click.style(f"Waiting for agents (timeout: {startup_timeout}s)...", fg="blue"))
+
+        try:
+            agent_cards, resolved_agent = await wait_for_agents(
+                url=url,
+                target_agent=agent,
+                token=token,
+                timeout=startup_timeout,
+                poll_interval=1.0,
+                stabilization_wait=2.0,
```
Comment (amir-ghasemi) [inline cli/commands/task_cmd/run.py:318]: should this be hardcoded to 2s?

File: cli/commands/task_cmd/artifact_handler.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            url = f"{self.base_url}/api/v1/artifacts/{self.session_id}/{filename}"
+
+        async with httpx.AsyncClient(timeout=60.0) as client:
+            response = await client.get(url, headers=self._get_headers())
+            response.raise_for_status()
+
+            content = response.content
+            mime_type = response.headers.get("content-type", "application/octet-stream")
+
+            local_path = self.artifacts_dir / filename
```
Comment (amir-ghasemi) [inline cli/commands/task_cmd/artifact_handler.py]: should we sanitize the filename?
